src/GHC/DataSize.hs

```diff
@@ -33,7 +37,7 @@ wORD_SIZE = SIZEOF_HSWORD
 
 -- | Calculate size of GHC objects in Bytes. Note that an object may not be
 --   evaluated yet and only the size of the initial closure is returned.
-closureSize :: Num b => a -> IO b
+closureSize :: a -> IO Word
```

Comment:
Yes, `Word` is basically like `size_t` in C. Its a type that can hold the number of bytes that can be addressed in a single process (assuming zero space for code).